### DISCUSSION TITLE
New Display, new try

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#20558] [Q&A]
Hello, after some non successfull tries with several displaysboards from AE, i got a new one. [ZX7D00CE01S-V13](https://www.aliexpress.com/item/1005006156603547.html?spm=a2g0o.order_list.order_list_main.11.61221802ml4Fho).

i found the datasheet for this board with google translate 
[ZX7D00CE01S.pdf](https://github.com/arendst/Tasmota/files/14003284/ZX7D00CE01S.pdf).

So i adjust the first line in `display.ini` to
`:H,ST7701,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,14`

But 

1) i have no idea what IC i have to choose. i choosed `ST7701` because, i found only one [example ](https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/ST7262_rgb16_display.ini ) with RGB interface and that has this IC set. On product page i can see ` Display driver IC: EK9716BD4`.
2) But in this [example ](https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/ST7262_rgb16_display.ini) the count of Header line paramter from behind 'RGB" is 22. In the documentation i do count 21 parameters after `RGB`. So i am very confused.
3) i have no information about `Pixel clock Speed in MHz (usually 14)`, i would try 14.

4) i tried the `[Quick Tour](https://tasmota.github.io/docs/HASPmota/)`. After downloading and rebooting i got this error:
`HSP: Error: OpenHASP demo requires PSRAM `
But this board should have PSRAM?!

5) i deleted this file again and tried the examples from [here ](https://tasmota.github.io/docs/LVGL_in_10_minutes/#try-a-lvgl-demo-app)and [here](https://tasmota.github.io/docs/LVGL/#using-lvgl).
pasting this in berry console i got this error:
```
BRY: Exception> 'value_error' - argument cannot be NULL
stack traceback:
	<native>: in native function
	<native>: in native function
	<unknown source>: in function `main`
```
i have no glue. i used this berry examples with other displays before (not successfully, but the berry code runs without error).

```
Programm Version:  13.3.0.1(lvgl-haspmota)
ESP Chip Id	13584184 (ESP32-S3 v0.1)
Flash Chip Id	0x18405E (QIO)
Flash Size	16384 KB
Program Flash Size	16384 KB
Program Size	2245 KB
Free Program Space	634 KB
Free Memory	216.5 KB
```
 
Could someone help?

**Comment by ChrisVonFrosta** [comment]
ok for 2) i was blind. `g0-g5 pins (green color)` has 6 bits

**Reply by sfromis** [reply]
For 1), the name you give is for identification, not for deciding function. Better stay close to what you actually have, which is `EK9716` (the rest is not chip type), instead of something random. And if not known, I'd give it as "unknown", instead.

**Comment by ameeuw** [comment]
Same with me - I also chose the wrong board type when recompiling. When using the online installer you can choose the "Tasmota32-S3 (QIO Flash/OPI PSRAM)" (lowest, in the unofficial section).

https://github.com/arendst/Tasmota/discussions/20527#discussioncomment-8183414

**Reply by ameeuw** [reply]
Short follow up:
According to the settings done here: https://github.com/fbiego/Windows_32/blob/11021812ed9397f6af0f14c4530593439196c84b/main/screen.c#L113-L161

The frequency should be 24MHz. I have updated the header line and added a `:V` section in order to set the parameters but it keeps on crashing as soon as I add this line.

```
:H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,24
:V,1,40,10,40,1,18,6,31,0
```

The hsync, vsync and pclk polarities are just guessed right now.

@gemu2015 I have added logging statements for the `:V` section and it takes over the values just fine, however somewhere between the V-handler and logging `Device: , xs:, ys: ...` we crash.

```
00:00:00.001 HDW: ESP32-S3 v0.1 (PSRAM)
00:00:00.040 UFS: FlashFS mounted with 12584 kB free
00:00:00.053 CFG: Loaded from File, Count 74
00:00:00.060 QPC: Count 1
00:00:00.061 SCR: ufilesystem found
00:00:00.112 BRY: Berry initialized, RAM used 5684 bytes
hsync_polarity  : 0
hsync_front_porch  : 40
hsync_pulse_width  : 10
hsync_back_porch  : 40
vsync_polarity  : 0
vsync_front_porch  : 18
vsync_pulse_width  : 6
vsync_back_porch  : 31
pclk_active_neg  : 1
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x40057143  PS      : 0x00060830  A0      : 0x8208712e  A1      : 0x3fcad070  
A2      : 0x00000000  A3      : 0x00000020  A4      : 0x0000003f  A5      : 0x3fcad030  
A6      : 0x3fcad030  A7      : 0xff000000  A8      : 0x00000000  A9      : 0x3fcacfd0  
A10     : 0x00000015  A11     : 0x3fcad030  A12     : 0x00000015  A13     : 0x3fcad070  
A14     : 0x3fcacff0  A15     : 0x00000008  SAR     : 0x00000015  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000000  LBEG    : 0x400556d5  LEND    : 0x400556e5  LCOUNT  : 0xffffffff  


Backtrace: 0x40057140:0x3fcad070 0x4208712b:0x3fcad090 0x42088377:0x3fcad220 0x42079029:0x3fcad260 0x4207f9c1:0x3fcad280 0x420a0aac:0x3fcad2a0 0x4207f9d9:0x3fcad360 0x420c08da:0x3fcad380 0x420cba48:0x3fcad3a0 0x420276c2:0x3fcad4e0
```

**Reply by gemu2015** [reply]
@ameeuw 
please post the complete descriptor.

also currently the pixel clock is limited to 14 Mhz internally.  While adapting my sunton display all values above 14 Mh lead to crashes.  if other displays work with higher frequencies we may remove that later.

**Reply by ChrisVonFrosta** [reply]
as you ask in the linked [thread](https://github.com/fbiego/Windows_32/blob/11021812ed9397f6af0f14c4530593439196c84b/main/screen.c#L113-L161) i was also thinking about testing more bits per pixel, because it seems to be more lanes according to this schmemtic

> Same with me - I also chose the wrong board type when recompiling. When using the online installer you can choose the "Tasmota32-S3 (QIO Flash/OPI PSRAM)" (lowest, in the unofficial section).

on another board i tried [here](https://github.com/arendst/Tasmota/discussions/20560), yesterday i ended with this binary and noticed that PSRAM shown up. 
On this board here, there comes no Serial connection if connecting to the PC but actually there is no usb/ttl adapter close to me.  I will change the binary asap with ttl/usb adapter.

would you share your values for `I:` parameters? i can't find the documentation.
i asume that is required to get some display output?!

**Reply by sfromis** [reply]
> On this board here, there comes no Serial connection if connecting to the PC

You may need to use a build for `tasmota32s3cdc` with the `CDC` part meaning to use USB CDC for serial via the built-in USB support of ESP32-S3.

**Reply by ChrisVonFrosta** [reply]
yes. i could update to build "tasmota32s3cdc"
even the same. no usb recognition on pc and no PSRAM. Display keep black/off/dark

```
00:00:00.002 CMD: Using USB CDC
00:00:00.003 HDW: ESP32-S3 v0.1 
00:00:00.035 UFS: FlashFS mounted with 12444 kB free
00:00:00.047 CFG: Loaded from File, Count 57
00:00:00.055 QPC: Count 1
00:00:00.114 BRY: Berry initialized, RAM used 4462 bytes
00:00:00.130 Project tasmota - Tasmota Version 13.3.0(tasmota32)-2_0_14(2023-12-12T14:31:55)
00:00:00.153 TAP: Loaded Tasmota App 'haspmota_demo.tapp'
00:00:00.561 HSP: Error: OpenHASP demo requires PSRAM
00:00:01.201 WIF: Connecting to AP1 
```

**Reply by sfromis** [reply]
I did not mean using that specific build, but one of the several builds starting with that, like the "[unofficial](https://github.com/tasmota/install/tree/firmware/firmware/unofficial)" `tasmota32s3cdc-qio_opi.bin`

**Reply by ChrisVonFrosta** [reply]
> I did not mean using that specific build, but one of the several builds starting with that, like the "[unofficial](https://github.com/tasmota/install/tree/firmware/firmware/unofficial)" `tasmota32s3cdc-qio_opi.bin`

with that binary i got error: "Upload failed - Invalid file signature

**Reply by sfromis** [reply]
I'm using that specififc build without problems. Do note that you have to find the URL for the downloadable "raw" file (no need to download it), do not just right-click + copy the URL you see on the linked page.

When it comes to Github and files, you should "never" assume that right-clicking on something gets you the "final" URL. You should generally first click through to a page for the file, and there you have buttons (including "raw") for the "real" URL to use.

**Reply by ameeuw** [reply]
@gemu2015 That was actually the full descriptor. I saw the `if (spi_freq>14)...` line so I set it to 14 to be sure. The board keeps on crashing for me and actually has some weird behaviour. 

But: I swear I saw some UI elements yesterday once: I had moved the `:V` section above the `:H` section. Then it worked for a split second and then it disappeared (literally faded away). After that I couldn't get it to work again...

**Comment by ChrisVonFrosta** [comment]
acutually it looks like:
![grafik](https://github.com/arendst/Tasmota/assets/16272493/79f76be3-54a5-49a4-8ad6-8b732d4a1d61)
https://github.com/arendst/Tasmota/assets/16272493/5200332e-fdb2-4d29-b7cb-cfc519641134

```
00:00:00.001 CMD: Using USB CDC
00:00:00.001 HDW: ESP32-S3 v0.1 (PSRAM)
00:00:00.037 UFS: FlashFS mounted with 12444 kB free
00:00:00.050 CFG: Loaded from File, Count 87
00:00:00.056 QPC: Count 1
00:00:00.059 SPI: Hardware using GPIO12(CLK), GPIO11(MOSI) and GPIO13(MISO)
00:00:00.060 I2C: Bus1 using GPIO20(SCL) and GPIO19(SDA)
00:00:01.007 TFS: File 'mcp23x.dat' not found
00:00:01.053 BRY: Berry initialized, RAM used 5684 bytes
00:00:01.173 TI: GT911 failed
00:00:01.742 DSP: Configured display 'EK9716BD4'
00:00:01.750 Project tasmota - Tasmota Version 13.3.0.4(lvgl-haspmota)-2_0_14(2024-01-22T17:44:03)
00:00:01.754 LVG: LVGL initialized
00:00:02.476 TAP: Loaded Tasmota App 'haspmota_demo.tapp'
00:00:04.252 BRY: failed to run compiled code 'value_error' - value needs to be a number
00:00:07.000 WIF: Connecting to AP1
```
My header line in `display.ini`
`:H,EK9716BD4,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,14`

![grafik](https://github.com/arendst/Tasmota/assets/16272493/a72d4cb4-d57a-436e-a98f-6c4df9b1df62)

![grafik](https://github.com/arendst/Tasmota/assets/16272493/01c0068a-e4d0-4ca0-b73e-97bfce6b0e40)

![grafik](https://github.com/arendst/Tasmota/assets/16272493/998f6e8d-83f3-4663-8131-384325e75df4)

but i found no informations for
```
:V
hsync_polarity,
hsync_front_porch,
hsync_pulse_width,
hsync_back_porch,
vsync_polarity,
vsync_front_porch,
vsync_pulse_width,
vsync_back_porch,
pclk_active_neg, 
```

**Reply by ameeuw** [reply]
> Short follow up: According to the settings done here: https://github.com/fbiego/Windows_32/blob/11021812ed9397f6af0f14c4530593439196c84b/main/screen.c#L113-L161
> 
> The frequency should be 24MHz. I have updated the header line and added a `:V` section in order to set the parameters but it keeps on crashing as soon as I add this line.
> 
> ```
> :H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,24
> :V,1,40,10,40,1,18,6,31,0
> ```

@ChrisVonFrosta  As suggested above, I took the values from the `Windows_32` demo.

**Reply by gemu2015** [reply]
this board does not need an initialization via spi.
however the lcd reset pin is wired to the io expander AW9523.  it has to be initialized properly before accessing the display.
 it is an i2c device and is not supported by universal driver.  may be that there is an option to initialize it via berry BEFORE universal display is called
would be helpful to have a working arduino driver source code

**Reply by gemu2015** [reply]
@ameeuw 

i am preparing an init mode in universal driver for this display, stay tuned

**Reply by gemu2015** [reply]
@ameeuw 

the special init is:
:II,B,DA
B = I2C Bus nr,  1 or 2
DA = hex device Adress of AW9523, normally 58
so probably with I2C bus 1
:II,1,58

then enter the opcodes for the AW9523 to set the output ports to the desired value
00,AA
means write address 00 with value AA
a single value means a delay
05 
means delay 5 milliseconds
 
; set both ports to direction output
04,00
05,00
; set output level to zero, except reset for display and touch
02,00
03,03
; reset is p10 of AW9523 means port 1 bit 0
; reset polarity not known, but seems to be active low
; set reset pins to 0
03,00
; delay 5 ms
5
; set reset pin to high again
03,03

would be much easier with working arduino source code

**Reply by ameeuw** [reply]
Tried it - thx for your quick work. Here's my report:

- Address lines of the AW9523 are both pulled high (according to schematic and code found here): https://github.com/fbiego/Windows_32/blob/11021812ed9397f6af0f14c4530593439196c84b/components/aw9523/aw9523.c#L15
- The init sequence of the AW9523 starts with a soft reset `7F,00`
- Your current code had some minor bugs in the debug print statements (`02%x´ instead of `%02x` and `%s` instead of `%d`).
- The reset init sequence can be found here: https://github.com/fbiego/Windows_32/blob/11021812ed9397f6af0f14c4530593439196c84b/main/screen.c#L97-L111

The following descriptor is what I brewed together. However this does not work consistently. I have seen a Tasmota logo (again!) and it was starting, then I started editing some values, I end up in a crash loop and have to remove option 3a, delete display.ini and then I can start from the beginning - it seems a little bewitched.

```
:H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,14
:II,1,5b
7f,00
64
04,00
05,00
02,00
03,03
64
03,00
64
03,03
64
:V,0,40,10,40,0,18,6,31,1
:0,00
:1,01
:2,02
:3,03
```

**Reply by gemu2015** [reply]
i tried the above display.ini on a blank esp32-s3 devkit and don't see any crashes.
that means the universal driver itself does not cause the crash (because all defined pins are output pins)
i guess a board incompatibility, probably PSRAM access.

**Reply by ameeuw** [reply]
Alright another _ghosty_ update. I have found the cause of the Kernel Panic: It is a space `" "` at the end of the last line. Just adding it to the end takes us one step further which yields this behaviour:

[ghostiness (video-converter.com).webm](https://github.com/arendst/Tasmota/assets/5405348/ceeb7612-bdde-4e17-92df-63447e4aa6d7)

The image is clearly showing in the beginning and then fading away slowly. @gemu2015 would like to get your input on this. I have also removed the 14MHz limit and tried it with 24MHz and observed the same behaviour.

Currently used descriptor:
```
:H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,14
:II,1,5b
7f,00
64
04,00
05,00
02,00
03,03
64
03,00
64
03,03
C8
:V,0,40,10,40,0,18,6,31,1
:B 64 02
:TI1,5d,*,*,-1,-1 
```

**Reply by gemu2015** [reply]
looks like the display bias voltages are not stable.  try to revert vsync polarity since it is an input to the bias voltage generator

:V,0,40,10,40,1,18,6,31,1

**Reply by ameeuw** [reply]
Nope - switching the polarities around didn't do the trick... we should probably look into the esp-idf's method of initialising an RGB panel, but I have the feeling your code is quite close to that.

**Reply by gemu2015** [reply]
look at the schematics, RGB_VS enables the bias generation and the converter chip is high enable.
do you have an oscilloscope ?  this pin must be high most of the time to work, so polarity should be 1

**Reply by gemu2015** [reply]
i just verified with oscilloscope VSYNC Polarity must be 1, else bias generator is off.

**Reply by gemu2015** [reply]
i never saw such a fading display. since we do not know the time constant of the R/C circuit on the enable pin of the booster. eventually it is not dependent on the polarity if the time constant is very high.  i would verify that the bias voltages are OK.
if they are, i don't have any guess what else might cause that problem.

**Reply by ChrisVonFrosta** [reply]
i remember i red that you compiled a custom build for testing your special init or ist that wrong? can't find it anymore.

with this init:
```
:H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,14
:II,1,5b
7f,00
64
04,00
05,00
02,00
03,03
64
03,00
64
03,03
C8
:V,0,40,10,40,0,18,6,31,1
:B 64 02
:TI1,5d,*,*,-1,-1 
```


i got this output:
```
00:00:00.001 CMD: Using USB CDC
00:00:00.001 HDW: ESP32-S3 v0.1 (PSRAM)
00:00:00.051 UFS: FlashFS mounted with 12444 kB free
00:00:00.063 CFG: Loaded from File, Count 110
00:00:00.069 QPC: Count 1
00:00:00.072 SPI: Hardware using GPIO12(CLK), GPIO11(MOSI) and GPIO13(MISO)
00:00:00.073 I2C: Bus1 using GPIO20(SCL) and GPIO19(SDA)
00:00:01.007 TFS: File 'mcp23x.dat' not found
00:00:01.060 BRY: Berry initialized, RAM used 5684 bytes
00:00:01.181 TI: GT911 failed
00:00:01.204 DSP: Configured display 'EK9716'
00:00:01.212 Project tasmota - Tasmota Version 13.3.0.4(lvgl-haspmota)-2_0_14(2024-01-22T17:44:03)
00:00:01.217 LVG: LVGL initialized
00:00:02.075 TAP: Loaded Tasmota App 'haspmota_demo.tapp'
00:00:03.889 BRY: failed to run compiled code 'value_error' - value needs to be a number
00:00:08.792 WIF: Connecting to AP1
```

and this result:
https://github.com/arendst/Tasmota/assets/16272493/d76a363c-f244-4e80-ba77-f67c14fb0c46

**Reply by ChrisVonFrosta** [reply]
with that `display.ini`:
```
:H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,24
:S,2,1,1,0,40,20
:V,1,40,10,40,1,18,6,31,0
:0,00
:1,01
:2,02
:3,03
:TI1,5d,*,*,-1,38
#
```

i git this output:
```
00:00:00.001 CMD: Using USB CDC
00:00:00.001 HDW: ESP32-S3 v0.1 (PSRAM)
00:00:00.056 UFS: FlashFS mounted with 12444 kB free
00:00:00.071 CFG: Loaded from File, Count 114
00:00:00.073 FRC: Some settings have been reset (2)
00:00:00.075 SPI: Hardware using GPIO12(CLK), GPIO11(MOSI) and GPIO13(MISO)
00:00:00.076 I2C: Bus1 using GPIO20(SCL) and GPIO19(SDA)
00:00:01.008 TFS: File 'mcp23x.dat' not found
00:00:01.063 BRY: Berry initialized, RAM used 5684 bytes
00:00:01.187 TI: GT911 failed
00:00:01.750 DSP: Configured display 'EK9716'
00:00:01.757 Project tasmota - Tasmota Version 13.3.0.4(lvgl-haspmota)-2_0_14(2024-01-22T17:44:03)
00:00:01.762 LVG: LVGL initialized
00:00:02.481 TAP: Loaded Tasmota App 'haspmota_demo.tapp'
00:00:04.345 BRY: failed to run compiled code 'value_error' - value needs to be a number
00:00:07.000 WIF: Connecting to AP1
```
and pretty the same display output as post before.

**Reply by ChrisVonFrosta** [reply]
> But: I swear I saw some UI elements yesterday once: I had moved the `:V` section above the `:H` section. Then it worked for a split second and then it disappeared (literally faded away). After that I couldn't get it to work again...

i countless restarts (and display.ini manipulations) i could also see the HASPmota demo for ~100ms  - only one time.

**Comment by ChrisVonFrosta** [comment]
> the special init is:....

i understand to less from your init. i will stay in spectator mode

**Comment by ChrisVonFrosta** [comment]
i found this [example code](https://github.com/smartpanle/QMSD-ESP32-BSP/tree/master/components/qmsd_board/board/esp32-s3/ZX7D00CE01S-8048) for this display!
Maybe that helps you smart guys to help ;)

**Comment by ChrisVonFrosta** [comment]
with this parameter for `:V`  i come closer and could see some HASPmota Demo artifacts for moments.
![grafik](https://github.com/arendst/Tasmota/assets/16272493/27ee0093-a8ea-49f3-9d70-eb1503e2177f)

```
:H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,24
:V,0,20,1,87,40,0,5,1,31,1
:II,1,5b
7f,00
64
04,00
05,00
02,00
03,03
64
03,00
64
03,03
C8
:B 64 02
:TI1,5d,*,*,-1,-1 
```
 see the video @second 30. HASPmota Demo pops up.

https://github.com/arendst/Tasmota/assets/16272493/84a7fc52-f002-4615-aa62-87a7f5f4ad25

this sequence is in a loop

**Comment by ChrisVonFrosta** [comment]
@ameeuw  try this `display.ini` 
```
:H,EK9716,800,480,16,RGB,39,38,5,9,45,17,16,15,7,6,21,0,46,3,8,18,10,11,12,13,14,14
:V,1,20,1,87,1,0,5,31,1

; :II,1,5B
;software reset
;7f,00
; ;enter the opcodes for the AW9523 ??? 00 iregister is read only!
; 00,AA
; ; set both ports to direction output
; 04,00
; 05,00
; ; set output level to zero, except reset for display and touch
; 02,00
; 03,03
; ; reset is p10 of AW9523 means port 1 bit 0
; ; reset polarity not known, but seems to be active low
; ; set reset pins to 0
; 03,00
; ; delay 5 ms
; 5
; ; set reset pin to high again
; 03,03
:B 64 02
:TI1,5d,*,*,-1,-1
```
i never come closer with this board:
https://github.com/arendst/Tasmota/assets/16272493/dd3ec79f-ae99-4afc-a41d-d77abcd9a6fe


in the post before i had one value to much (40).
Adjusting  `:V`  AND changing the polarity brings massive effect
The `:II 1,5b` command doesn't effect the result in anyway. (i tried all adresses)

i do not understand this.  00H register is read only.
```
; ;enter the opcodes for the AW9523 ??? 00 iregister is read only!
; 00,AA
```

i can see the Tasmota boot screen, flickering and the HASPmota startpage. Sometime it is "glitching". Sometimes the image "brake together" and comes back. Touch still not working.
Think that is caused in the missing or not working `:II` instruction and missing instuctions for the port expander AW9523 ?!

for aw9523 reset logic:
![grafik](https://github.com/arendst/Tasmota/assets/16272493/fc6b309e-c4ae-40fe-97f4-9e92795a779e)

**Reply by ameeuw** [reply]
Well done! Great find of that additional working example. Except for the least line of the display being a twitchy, it works quite well.

Your touch does not work as you have to set I2C SCL (47) & SDA (48) in the "Configure Module" GUI in Tasmota. The asterisk `"*"` in the descriptor denotes that the pins designated by the configuration are taken over. Alternatively you can set the pins in the descriptor line: `:TI1,5d,47,48,-1,-1`.

@gemu2015 Thank you for checking the polarity/fading issues. I assume those `:V` values were plain wrong. The new sample code also lists a pclk frequency of 24MHz - could that be the reason for the last line of the display being twitchy? Do the `:V` pulse widths need to be adjusted if we change the frequency? in this case multiply them by 24/14 in order to have the same duty cycles?

![IMG_5152](https://github.com/arendst/Tasmota/assets/5405348/ca7b72fb-a625-4b05-91f0-c340eabb20c1)

**Reply by gemu2015** [reply]
@ameeuw @ChrisVonFrosta 
ok we come close. BTW in your descriptor files you forgot the termination character '#'
this may crash the driver in some cases.
the last white display line is an issue of a still wrong parameter in :V,  check the 3 VSYNC parameters.
since i think it is no longer a driver but parameter issue i will do a pr for the current changes.

**Reply by ChrisVonFrosta** [reply]
> Your touch does not work as you have to set I2C SCL (47) & SDA (48) in the "Configure Module" GUI in Tasmota. The asterisk >`"*"` in the descriptor denotes that the pins designated by the configuration are taken over. Alternatively you can set the pins in >the descriptor line: `:TI1,5d,47,48,-1,-1`. 

i know and tried both ways. i had set i2c in configuration many times. But allways get the Error:
 `00:00:01.145 TI: GT911 failed`
What's overlooked? Configuration and display.ini should be all?
Is your touch working? would you share your hole corresponding setttings/configuration in total for once?


one confusing thing:
i did the setup from `:V`  belong to this:
![299827976-27ee0093-a8ea-49f3-9d70-eb1503e2177f](https://github.com/arendst/Tasmota/assets/16272493/91bd0411-957b-47ac-8ae3-0fe9fac68710)
it should result in:
`:V,1,20,1,87,1,5,1,31,1`

but luckiLy i did a mistake and we use now:
`:V,1,20,1,87,1,0,1,31,1`

after noticed (latenight) that , i tried the "correct" values and got crashing screen again
Even the polarity seems to be not most critical parameter.
The most effectiv/critical parameter seems to be the "0" for `vsync_front_porch,`

**Reply by ChrisVonFrosta** [reply]
i was also unsure if the white line at the bottom is "normal" (shinig backlight) or a settings issue.

**Reply by ChrisVonFrosta** [reply]
sometimes editing any file in webgui do effect to the Screen in glitching or broken image

**Comment by ChrisVonFrosta** [comment]
@ameeuw would you share your files/configs?
i have still the "broken" line at the bottom.
When programm reacts and goes two screen2 the last line becomes correctly.
After a short wait period, screen1 is loaded again and last last becomes "broke" again.

**Reply by ameeuw** [reply]
@ChrisVonFrosta No updates so far. I have looked at the data sheet and found some details on the timing, saying that we need a VSYNC Front Porch of minimum 1. But as soon as I set that to one, the Bias voltage (I assume) goes sideways and the "fading" of the display is back.

**Reply by ChrisVonFrosta** [reply]
So you also have still a bad line at the bottom?

I asked the seller/manufacturer and await the anwser - hopfully a usefull one.

In the meanwhile:
Is there an option to measure the "sync signal" with an oscilloscope? To see the effekt of tweking or to see which value is to adjust or in which direction?
